psl/psl.go

```diff
@@ -0,0 +1,171 @@
+// package psl defines the Psl struct containing alignment data as well as functions and methods that operate on the Psl struct
+package psl
+
+import (
+	"fmt"
+	"github.com/vertgenlab/gonomics/common"
+	"github.com/vertgenlab/gonomics/fileio"
+	"strings"
+)
+
+// Psl struct declares data fields and types for Psl struct.
+type Psl struct {
+	Match       int
+	MisMatch    int
+	RepeatMatch int
+	Ns          int
+	QNumIns     int
+	QBaseIns    int
+	TNumIns     int
+	TBaseIns    int
+	Strand      string
+	QName       string
+	QSize       int
+	QStart      int
+	QEnd        int
+	TName       string
+	TSize       int
+	TStart      int
+	TEnd        int
+	BlockCount  int
+	BlockSize   []int
+	QList       []int
+	TList       []int
+}
+
+// PslReader is a struct that contains a SimpleReader with an embedded bufioReader and a 2-D byte slice to reduce memory allocation when reading each line.
+type PslReader struct {
+	Reader  *fileio.SimpleReader
+	curr    Psl
+	columns []string
+	err     bool
+}
+
+// NewPslReader will open a given text file and return a pointer to a PslReader struct.
+func NewPslReader(filename string) *PslReader {
+	var fields []string
+	var done bool
```

Comment:
Are these variables outside the struct used?  Seems like they just hold default values, which are then assigned to the variables within the struct.  For example, could this line/variable be removed then then 'false' put for the initial value of err in the struct?

Reply:
You have a good point after thinking about this I dont think these variable have that much upside.